shared/constants/permissions.ts

```diff
@@ -31,8 +31,43 @@ export const RestrictedMethods = Object.freeze({
 export const ConnectionPermission = Object.freeze({
   connection_permission: 'connection_permission',
 });
+
+// This configuration specifies permission weight thresholds used to determine which
+// permissions to show or hide on certain Snap-related flows (Install, Update, etc.)
+export const PermissionWeightThreshold = Object.freeze({
+  snapInstall: 3 as const,
+  snapUpdateApprovedPermissions: 2 as const,
+});
 ///: END:ONLY_INCLUDE_IF
 
+export const PermissionWeight = Object.freeze({
```

Comment:
Since we're defining the weights for each permission here, it might be worth it to update `PERMISSION_DESCRIPTIONS`  in `ui/helpers/utils/permission.js` to use this object to ensure that this is the single source of truth.